src/components/dialog/dialog.ts

```diff
@@ -116,27 +121,43 @@ export default class IgcDialogComponent extends EventEmitterMixin<
   }
 
   protected override async firstUpdated() {
+    this.animationPlayer = new AnimationPlayer(this.dialog);
     await this.updateComplete;
     if (this.open) {
       this.dialog.showModal();
     }
   }
 
+  private async toggleAnimation(dir: 'open' | 'close') {
+    const animation = dir === 'open' ? fadeIn : fadeOut;
+
+    const [_, event] = await Promise.all([
+      this.animationPlayer.stopAll(),
+      this.animationPlayer.play(animation),
+    ]);
+
+    return event.type === 'finish';
+  }
+
   /** Opens the dialog. */
   public show() {
     if (this.open) {
       return;
     }
 
+    this.backdrop.setAttribute('aria-hidden', 'false');
```

Comment:
Why do we need to have this here and elsewhere, when it's already being set in the template by binding?